### PR TITLE
Add idna encode/decode helpers

### PR DESCRIPTION
**Pull request by ross**
/cc https://github.com/octodns/octodns/issues/199
/cc https://datatracker.ietf.org/doc/html/rfc3490

### CODE REVIEW

File: octodns/idna.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        if name.startswith('*'):
+            # idna.encode doesn't like the *
+            name = _encode(name[2:]).decode('utf-8')
+            return f'*.{name}'
+        return _encode(name).decode('utf-8')
+
+
+def idna_decode(name):
+    pieces = name.lower().split('.')
+    if any([p.startswith('xn--') for p in pieces]):
```
Comment (viranch) [inline octodns/idna.py:25]: Slightly better?

```suggestion
    if any(p.startswith('xn--') for p in pieces):
```

Comment (ross) [reply octodns/idna.py:25]: :cool: guess I didn't know the list comprehension would work w/o the `[]`

**Comment by viranch** [comment]
> Each provider will need updates to pull in and use these methods appropriately at the last minute.

Some documentation/hints around how and when to use the helpers would be very helpful for the providers.

File: octodns/idna.py (hunk 1 above)
Comment (viranch) [reply octodns/idna.py]: I think in my version, `any` will get an iterator which will stop evaluating subsequent elements after it finds a `True`.

**Comment by ross** [comment]
> Some documentation/hints around how and when to use the helpers would be very helpful for the providers.

To some degree that's still TBD, but I can probably put something together based on what NS1 looks like https://github.com/octodns/octodns-ns1/pull/20.

**Comment by viranch** [comment]
> I can probably put something together based on what NS1 looks like

Even a PR would be a good starting point for reference. Doesn't have to be in this PR.

**Review by viranch — APPROVED** [review]
Simple enough from code perspective, I don't have use-cases irl to test it. Ship it!

**Comment by ross** [comment]
After getting distracted and working on non-octoDNS stuff for a bit I'm getting back to this.

**Pull request merged**